core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -52,20 +52,38 @@ public abstract class ManifestWriter<F extends ContentFile<F>> implements FileAp
   private long deletedRows = 0L;
   private Long minDataSequenceNumber = null;
 
-  private ManifestWriter(PartitionSpec spec, OutputFile file, Long snapshotId) {
+  private ManifestWriter(
+      PartitionSpec spec,
+      OutputFile file,
+      Long snapshotId,
+      String compressionCodec,
+      Integer compressionLevel) {
     this.file = file;
     this.specId = spec.specId();
-    this.writer = newAppender(spec, file);
+    this.writer = newAppender(spec, file, compressionCodec, compressionLevel);
     this.snapshotId = snapshotId;
     this.reused = new GenericManifestEntry<>(spec.partitionType());
     this.stats = new PartitionSummary(spec);
   }
 
   protected abstract ManifestEntry<F> prepare(ManifestEntry<F> entry);
 
+  /**
+   * @deprecated since 1.4.0, will be removed in 1.5.0; use {@link
+   *     ManifestWriter#newAppender(PartitionSpec, OutputFile, String, Integer)} instead.
+   */
+  @Deprecated
   protected abstract FileAppender<ManifestEntry<F>> newAppender(
       PartitionSpec spec, OutputFile outputFile);
 
+  protected FileAppender<ManifestEntry<F>> newAppender(
```

Comment:
Should this be an abstract and use this as the default implementation for the above method?

Reply:
I'm not sure I understand your comment. I'm keeping the existing method abstract so as not to break the API. This one has an implementation, so it cannot be abstract.

Reply:
The `newAppender(PartitionSpec spec, OutputFile outputFile);` is marked as deprecated. This method still needs to be modified to `abstract` after the `newAppender(PartitionSpec spec, OutputFile outputFile)` is removed in 1.5.0. However, it will break the API.

Reply:
@ConeyLiu I see your point.
@nastra recommended making the new `newAppender` with 4 parameters non-abstract (calling the old `newAppender` with 2 parameters) to avoid breaking the API. However, as you point out, when we remove the deprecated 2-parameter `newAppender` in a future release, this 4-parameter `newAppender` will have to become abstract, which will break the API then. We either break the API now or later. @nastra what do you think?